sys/xtimer/xtimer.c

```diff
@@ -233,8 +233,11 @@ static void _mutex_timeout(void *arg)
     mutex_thread_t *mt = (mutex_thread_t *)arg;
 
     mt->timeout = 1;
-    sched_set_status(mt->thread, STATUS_PENDING);
     list_remove(&mt->mutex->queue, (list_node_t *)&mt->thread->rq_entry);
+    if (mt->mutex->queue.next == NULL) {
```

Comment:
Really good catch! :+1: 

I ignored that the `MUTEX_LOCK` is not always the tip of the list! It is only when the mutex is owned by one thread. If there are more threads waiting, the tip is NULL. I guess this is done to be able to use `thread_add_to_list` cleanly.

There's one race condition missing. Otherwise if the timer fires between setting the timer and locking the mutex might create inconsistencies in the mutex queue. Just test that `list_remove` really removed a node

```C
list_node_t* node = list_remove(...)
if (node != NULL && mt->mutex->queue.next == NULL) {
    // we were the last node in the queue but the mutex shall still be owned by someone else
    mt->mutex->queue.next = MUTEX_LOCKED;
}
```